### PR TITLE
CMake samples cleanup

### PR DESCRIPTION
**Pull request by andschwa**
Re-opening #1638 manually (which replaced #1618) because I force-pushed the branch.

### CODE REVIEW

**Comment by andschwa** [comment]
bors try

**Comment by andschwa** [comment]
bors try

**Comment by andschwa** [comment]
bors try

**Comment by andschwa** [comment]
Samples are documentation by example, so this is doc-a-thon worthy!

**Comment by andschwa** [comment]
bors try

**Comment by andschwa** [comment]
Hm, one more kink to iron out. The new way is failing to build with the old Makefiles. @aserdean feel free to take a look, but please add on this this branch, I've added a couple commits we definitely need (e.g. progress was made!)

**Comment by aserdean** [comment]
bors try

**Comment by aserdean** [comment]
bors try

**Comment by aserdean** [comment]
@andschwa I modified a few things can you PTAL?

I also got a successful run now.

Is it safe to remove https://github.com/Microsoft/openenclave/blob/master/samples/run also?

**Comment by andschwa** [comment]
@aserdean regarding that last commit, I think we're both right. We still need to use the installed pkg-config (to test that it deployed correctly), but we also need OE_PREFIX set. 😆

**Comment by andschwa** [comment]
Yes let's remove `run` too. Nice work, by the way.

**Comment by aserdean** [comment]
Removed.

**Comment by andschwa** [comment]
bors try

File: samples/test-samples.cmake [hunk 1]
```diff
@@ -50,7 +50,7 @@ foreach (SAMPLE data-sealing file-encryptor helloworld local_attestation remote_
 
     # Build with pkg-config
     execute_process(
-      COMMAND sh -c "PATH=${INSTALL_DIR}/bin:$PATH PKG_CONFIG_PATH=${BUILD_DIR}/pkgconfig OE_PREFIX=${INSTALL_DIR} make -C ${SAMPLE_SOURCE_DIR} clean build run"
+      COMMAND ${CMAKE_COMMAND} -E env PATH=${INSTALL_DIR}/bin:$ENV{PATH} PKG_CONFIG_PATH=${BUILD_DIR}/pkgconfig OE_PREFIX=${INSTALL_DIR} COMMAND make -C ${SAMPLE_SOURCE_DIR} clean build run
```
Comment (aserdean) [inline samples/test-samples.cmake]: execute_process `COMMAND` is executed in [parallel](https://cmake.org/cmake/help/v3.12/command/execute_process.html) we could use add_custom_command [instead](https://cmake.org/cmake/help/v3.12/command/add_custom_command.html#command:add_custom_command)

**Comment by andschwa** [comment]
bors try

File: samples/test-samples.cmake (hunk 1 above)
Comment (andschwa) [reply samples/test-samples.cmake]: That's only if `COMMAND COMMAND COMMAND` is used; we're only using one command for each execute_process so this doesn't affect us.

...

And that's a typo on my part, that COMMAND doesn't belong, I was blindly following the man page for CMake in command line mode. Thanks!

**Comment by andschwa** [comment]
bors try

**Comment by andschwa** [comment]
Oh no @aserdean I see why you were using the built pkgconfig files, not the installed ones.

I changed it back because we purposefully need to test that the install is relocatable...but apparently it's not! The existing pkgconfig setup is 100% not relocatable with DESTDIR. But I'm not going to fix this right now, so I need to file a new issue.

**Comment by andschwa** [comment]
bors try

File: samples/test-samples.cmake [hunk 2]
```diff
@@ -71,6 +72,7 @@ foreach (SAMPLE data-sealing file-encryptor helloworld local_attestation remote_
     endif ()
 
     # Build with pkg-config
+    # TODO: Use INSTALL_DIR for pkg-config after it is made relocatable.
     execute_process(
       COMMAND ${CMAKE_COMMAND} -E env PATH=${INSTALL_DIR}/bin:$ENV{PATH} PKG_CONFIG_PATH=${INSTALL_DIR}/share/pkgconfig OE_PREFIX=${INSTALL_DIR} make -C ${SAMPLE_SOURCE_DIR} clean build simulate
```
Comment (aserdean) [inline samples/test-samples.cmake]: `PKG_CONFIG_PATH=${BUILD_DIR}/`

Comment (andschwa) [reply samples/test-samples.cmake]: Oof, we should de-duplicate if we can (later), this is like the third time I've made the correct change to one line and not the other. Thanks.

**Comment by andschwa** [comment]
bors try

**Comment by aserdean** [comment]
> Oh no @aserdean I see why you were using the built pkgconfig files, not the installed ones.
> 
> I changed it back because we purposefully need to test that the install is relocatable...but apparently it's not! The existing pkgconfig setup is 100% not relocatable with DESTDIR. But I'm not going to fix this right now, so I need to file a new issue.

TBH I thought it was a bit odd, but I guessed that was the intent as per: https://github.com/Microsoft/openenclave/blob/master/samples/test-samples.sh#L41

**Pull request merged**